tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/JoinClauseTest.java

```diff
@@ -120,6 +120,12 @@ public void testOuterJoinWithNullResults() {
         BRunUtil.invoke(result, "testOuterJoin");
     }
 
+    @Test(description = "Test join clause with a large list")
+    public void testJoinClauseWithLargeList() {
+        Object values = BRunUtil.invoke(result, "testJoinClauseWithLargeList");
+        Assert.assertTrue((Boolean) values);
```

Comment:
Shall we assert this in `join-clause.bal` itself.

Reply:
Done.